### PR TITLE
Fix notary request handling

### PR DESCRIPTION
**Pull request by roman-khimov**

### CODE REVIEW

File: pkg/morph/event/notary_preparator.go [hunk 1]
```diff
@@ -141,15 +143,10 @@ func (p preparator) Prepare(nr *payload.P2PNotaryRequest) (NotaryEvent, error) {
...
-	// or have it filled with dummy bytes (if request was created manually with the old
-	// neo-go API)
-	//
-	// this check prevents notary flow recursion
-	if !(len(nr.MainTransaction.Scripts[1].InvocationScript) == 0 ||
-		bytes.Equal(nr.MainTransaction.Scripts[1].InvocationScript, p.dummyInvocationScript)) { // compatibility with old version
+	// We do receive requests made by the node itself, but these must never
+	// be reacted upon, otherwise we'll never end sending these requests
+	// (at least not until the main tx is ready).
+	if nr.FallbackTransaction.Signers[1].Account == p.localAcc {
```
Comment (cthulhu-rider) [inline pkg/morph/event/notary_preparator.go:149]: i'd insert a remark that 2nd signer obliged to attend according to https://pkg.go.dev/github.com/nspcc-dev/neo-go@v0.105.0/pkg/rpcclient#WSClient.ReceiveNotaryRequests. Code will look safer with it

Comment (cthulhu-rider) [inline pkg/morph/event/notary_preparator.go:149]: https://pkg.go.dev/github.com/nspcc-dev/neo-go@v0.105.0/pkg/util#Uint160.Equals is more preferable way when provided

**Review by cthulhu-rider — APPROVED** [review]

File: pkg/morph/event/listener.go [hunk 2]
```diff
@@ -43,12 +43,13 @@ type Listener interface {
...
-	// notary mainTX signer. In practise, it means that listener will subscribe
-	// for only notary requests that are going to be paid with passed hash.
+	// EnableNotarySupport enables notary request listening. Passed hashes are
+	// local key account and notary mainTX signer. In practice, it means that listener
+	// will subscribe only for notary requests that are going to be paid with passed
+	// mainTX hash.
 	//
 	// Must not be called after Listen or ListenWithError.
-	EnableNotarySupport(util.Uint160, client.AlphabetKeys, BlockCounter)
+	EnableNotarySupport(util.Uint160, util.Uint160, client.AlphabetKeys, BlockCounter)
```
Comment (carpawell) [inline pkg/morph/event/listener.go:52]: let's name args since they are same-typed now?

**Review by carpawell — COMMENTED** [review]
- Conflicts
- TBH, I would wait for https://github.com/nspcc-dev/neo-go/issues/3288

Otherwise, I am OK.

**Review by cthulhu-rider — APPROVED** [review]

**Review by carpawell — APPROVED** [review]

**Pull request merged**